### DISCUSSION TITLE
Create a new child record automatically.

### DISCUSSION THREAD

**Discussion by gvlx** [Budibase/budibase#3351] [Help]
Hi,

Following #3223 and #2043 , and working around #2054, I'm trying to automatically create a new child record of a main record.

As soon as the main record is created I want to inject the main record's ID and on use it to create the new child record, without an user intervention.

Is there a way to do this without opening a new page for the child's records?

Thanks for the support (and for creating `bb`).

**Comment by gvlx** [comment]
Automation seemed like the way to go but there is no way to set the relationship field there.

new row `test_1` automatically creates a new row `child of test_1`:
![imagem](https://user-images.githubusercontent.com/67311/141472092-c640361b-a00a-44b8-a0c1-8570478305e3.png)
![imagem](https://user-images.githubusercontent.com/67311/141472131-860e1a8d-5528-46ed-abe4-3e904bdb2652.png)

Using this automation:
![imagem](https://user-images.githubusercontent.com/67311/141472261-fcc584ba-8823-4d0a-8543-44bdf733ab50.png)

But because of #3344 the child is left orphaned.
Any ideas for a workaround?

**Comment by gvlx** [comment]
If using a form like this:

![imagem](https://user-images.githubusercontent.com/67311/141473189-2429e84d-fca3-4bbb-ae3d-0cf0cfb958af.png)

Is there a way to 'press' the `New child` button when entering the page? Or just execute its actions?

**Comment by mjashanks** [comment]
Does this help at all: https://docs.budibase.com/guides/create-a-child-record-from-the-parent ?

**Comment by gvlx** [comment]
Hi, I've used it to create that example.

Now, If I could only make the child without the user's input....
